### ISSUE TITLE
Snap + Snap-Zoom don't work correctly

### ISSUE THREAD

**Issue by SukantPal** [davidfig/pixi-viewport#265]
**TLDR;** When using snap + snap-zoom, the "visible" frame doesn't move as expect. The viewport zooms in at a different rate than it moves its center.

If animating between two rectangles (in case of viewport, b/w two getVisibleBounds()), the interpolation parameter _t_ should be the same in each frame.

<img width="397" alt="Screen Shot 2020-10-17 at 10 33 50 AM" src="https://user-images.githubusercontent.com/22450567/96339682-46785f80-1064-11eb-9f66-f653a60ef231.png">

The **width and height** need to be interpolated with the same parameter _t_ as the position **x and y**. This can be proved as follows:

<img width="754" alt="Screen Shot 2020-10-17 at 10 45 37 AM" src="https://user-images.githubusercontent.com/22450567/96339963-e97da900-1065-11eb-876c-796a49ce82a6.png">

**However, snap zoom interpolates the initial and final scales. Since the width/height and scales have an inverse relationship - the animation is off**

----
A better intuition of this problem: let's animate a rectangle from (0, 0) to (1, 1) and dimensions from (1, 1) to (1/2, 1/2). Let's assume the base width/height is (1, 1), so that the initial zoom is 1 and the final zoom is 2.

* The blue line is the animated position.
* The green line is the animated **scale**.
* The purple curve is the animated **width and height**.

Notice how the scale is linear but the width is not. The width curve's derivative is changing, and so it feels like the animation is off. Furthermore, using "linear" ease on the snapZoom plugin **won't feel like a linear animation**.

<img width="1325" alt="Screen Shot 2020-10-17 at 11 06 51 AM" src="https://user-images.githubusercontent.com/22450567/96346291-e46e2900-1068-11eb-8db3-4b60db3b4ed7.png">

---
Solution: The snap zoom plugin should interpolate between the initial/final width/height of the viewport's visible bounds. The scale then should be calculated from that instead.

> The fundamental cause of this anomaly is that scale and visible-bound's dimensions are inversely related. If they were directly related, then there would've not been any problem. Increasing the visible's bounds by a width _dw_, doesn't cause a proportional change in the scale _ds_. Rather _dw / w_ =  _s_ / _ds_.

**Comment by SukantPal** [comment]
I'm leaving the changes I made to fix SnapZoom here:

```
// L94
createSnapping() {
        const scale = this.parent.scale
        const worldScreenWidth = this.parent.worldScreenWidth
        const worldScreenHeight = this.parent.worldScreenHeight
        const destScreenWidth = this.parent.screenWidth / this.xScale
        const destScreenHeight =  this.parent.screenHeight / this.yScale

        this.snapping = {
          time: 0,
          startX: worldScreenWidth,
          startY: worldScreenHeight,
          deltaX: destScreenWidth - worldScreenWidth,
          deltaY: destScreenHeight - worldScreenHeight
        }
        this.parent.emit('snap-zoom-start', this.parent)
    }
```

```
// inside update() -> else if (this.snapping) -> else {

            else {
                const snapping = this.snapping
                const width = this.ease(snapping.time, snapping.startX, snapping.deltaX, this.options.time)
                const height = this.ease(snapping.time, snapping.startY, snapping.deltaY, this.options.time)

                this.parent.scale.x = this.parent.screenWidth / width
                this.parent.scale.y = this.parent.screenHeight / height
            }
```

**Comment by davidfig** [comment]
Take a look at `viewport.animate()`. I ran into this a while ago and created a new plugin to handle it without breaking old plugin functionality. Do you think it's worth revisiting it to fix the original behavior?

**Comment by SukantPal** [comment]
@davidfig Yes. It doesn't take a lot of changes to fix this. I can make a PR if needed.

**Comment by davidfig** [comment]
Cool. If you wouldn't mind, I know a bunch of people ran into this problem, and it would be great if the plugins worked well together.
